website/docs/d/entity.html.markdown

```diff
@@ -165,12 +165,66 @@ In addition to all arguments above, the following attributes are exported:
 * `guid` - The unique GUID of the entity.
 * `application_id` - The domain-specific application ID of the entity. Only returned for APM and Browser applications.
 * `serving_apm_application_id` - The browser-specific ID of the backing APM entity. Only returned for Browser applications.
+* `entity_tags` - A JSON-encoded string, comprising tags associated with the entity fetched. 
+   * See the **Additional Examples** section below, for an illustration depicting the usage of `jsondecode` with the attribute `entity_tags`, to get the tags associated with the entity fetched.
 
 
 ## Additional Examples
 
 -> If the entities are not found please try again without providing the `type` field.
 
+### Entity Tags
+
+* The `entity_tags` attribute allows you to retrieve the tags associated with a New Relic entity. Tags are key-value pairs that help categorize and identify entities based on various criteria such as environment, team, project, etc.
+* Consider a scenario where you want to retrieve an entity and use its tags. Here's how you can access and process the `entity_tags` attribute.
+  
+  `data.newrelic_entity.foo.entity_tags` accesses the `entity_tags` attribute.
+  `!= null` checks if the `entity_tags` attribute is not null.
+  `jsondecode(data.newrelic_entity.foo.entity_tags)` decodes the JSON string into a list of objects.
+  `{ for pair in jsondecode(data.newrelic_entity.foo.entity_tags) : pair.key => pair.values }` iterates over each object in the decoded list and constructs a map where each `pair.key` is mapped to `pair.values`.
+
+Here's an example of how to use this in a Terraform configuration:
+
+```hcl
+data "newrelic_entity" "foo" {
+  name = "Dummy App - Late'ncy"
+  domain = "EXT"
+  type = "SERVICE_LEVEL"
+}
+locals {
+  key_value_map = data.newrelic_entity.foo.entity_tags != null ? { for pair in jsondecode(data.newrelic_entity.foo.entity_tags) : pair.key => pair.values } : null
+}
+
+output "key_value_map" {
+  value = local.key_value_map
+}
+
+```
+Let's consider an example where the entity has the following tags:
+
+```hcl
+env: production
+team: ops
+```
+
+In JSON format, the entity_tags might look like this:
+
+```hcl
+[
+  {"key": "env", "values": ["production"]},
+  {"key": "team", "values": ["ops"]}
+]
+```
+
+The `local.key_value_map` would be:
+
+```hcl
+{
+  env = ["production"]
+  team = ["ops"]
+}
+```
+
```

Comment:
```suggestion
### Using the `entity_tags` Attribute to Fetch Tags Associated with the Entity

As stated above in the **Attribute Reference** section, while the attribute `entity_tags` helps retrieve tags associated with the entity fetched by the data source, the tags are returned as a JSON-encoded string and not a conventional list or a map, owing to a couple of design considerations; which is why one would need to use the Terraform function `jsondecode()`, along with the attribute `entity_tags` in order to convert the JSON-encoded string into a map with key-value pairs. 

The following is an illustration of the aforementioned scenario. It may be observed that a key-value pair version of the JSON-encoded string exported by `entity_tags` is written to the variable `key_value_maps` , using the `jsondecode()` function.

```hcl
data "newrelic_entity" "foo" {
  name = "Sample Searchable Entity"
  domain = "EXT"
  type = "SERVICE_LEVEL"
}

locals {
  key_value_map = { for pair in jsondecode(data.newrelic_entity.foo.entity_tags) : pair.key => pair.values }
}

output "key_value_map" {
  value = local.key_value_map
}
```
The value of `local.key_value_map`  would look like the following.
```hcl
{
  env = ["production"]
  team = ["ops"]
}
```

In order to prevent an error being thrown by the variable added if no tags are associated with the entity (consequently, the value of `entity_tags` is an empty string), the value of the variable may be conditionally defined in order to not throw an error, as shown below.

```hcl
locals {
  key_value_map = data.newrelic_entity.foo.entity_tags != null ? { for pair in jsondecode(data.newrelic_entity.foo.entity_tags) : pair.key => pair.values } : null
}
```

```

@abasha1234 can you please take a look at this change, commit this, and shift this section below all of the sections we have under **Additional Examples**?

Fantastic work on the explanation you've written, but this would be a bit too lengthy/elaborate, and we might want to keep this concise to just help the customer understand they would need to use `jsondecode()` with `entity_tags`; hence these changes :)

Reply:
Hey @pranav-new-relic thanks for feedback.